ksql-common/src/test/java/io/confluent/ksql/function/UdfFactoryTest.java

```diff
@@ -25,29 +25,105 @@
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 
+import java.util.Arrays;
+import java.util.Collections;
+
 import io.confluent.ksql.function.udf.Kudf;
 import io.confluent.ksql.function.udf.UdfMetadata;
+import io.confluent.ksql.util.KsqlException;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
 
 public class UdfFactoryTest {
 
   @Rule
   public final ExpectedException expectedException = ExpectedException.none();
 
   private UdfFactory factory;
+  private String functionName;
 
   @Before
   public void setUp() {
-    factory = new UdfFactory(TestFunc.class, new UdfMetadata("TestFunc", "", "", ""));
+    functionName = "TestFunc";
+    factory = new UdfFactory(TestFunc.class, new UdfMetadata(functionName, "", "", ""));
   }
 
   @Test
   public void shouldThrowIfNoVariantFoundThatAcceptsSuppliedParamTypes() {
     expectedException.expect(KafkaException.class);
     expectedException.expectMessage("Function 'TestFunc' does not accept parameters of types:[VARCHAR(STRING), BIGINT]");
 
-    factory.getFunction(ImmutableList.of(Schema.Type.STRING, Schema.Type.INT64));
+    factory.getFunction(ImmutableList.of(Schema.STRING_SCHEMA, Schema.INT64_SCHEMA));
+  }
+
+  @Test
+  public void shouldFindFirstMatchingFunctionWhenNullTypeInArgs() {
+    final KsqlFunction expected = new KsqlFunction(Schema.STRING_SCHEMA,
+        Collections.singletonList(Schema.STRING_SCHEMA),
```

Comment:
Schema.STRING_SCHEMA in the param list would _always_ be optional, right?  Cos its coming from SchemaUtil.typeToSchema...

(Other tests need changing too)